charts/hook/templates/post-delete.yaml

```
@@ -0,0 +1,35 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "app.fullname" . }}-cleanup
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+  annotations:
+    argocd.argoproj.io/hook: PostDelete
+    argocd.argoproj.io/hook-delete-policy: HookSucceeded
+spec:
+  template:
+    metadata:
+      labels:
+        {{- include "app.labels" . | nindent 8 }}
+        job-name: {{ include "app.fullname" . }}-cleanup
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: cleanup
+        image: {{ .Values.appTypesImage.repository }}:{{ .Values.appTypesImage.tag }}
+        imagePullPolicy: {{ .Values.appTypesImage.pullPolicy }}
+        args:
+          - "cleanup"
+          - "{{ .Values.cleanupArgs.app_type }}"
+          - "{{ .Values.cleanupArgs.app_id }}"
+          - "--output-type"
+          - "{{ .Values.cleanupArgs.output_type_name }}"
+          - "--package-name"
+          - "{{ .Values.cleanupArgs.package_name }}"
```

1. What can we do with older app versions, which app-types do not have `cleanup` command?

2. This helm hook [will be called](https://github.com/neuro-inc/platform-apps/blob/master/resources/workflow.yaml#L323-L328) even before platform-apps is being released with updated argo-workflow.

Let's add switch to enable this hook that preprocessor will enable + let's allow it to fail if command is not found, wdyt?

---

@YevheniiSemendiak this hook doesn't do anything if it fails. It's a Post Delete hook that tries to run itself 5 times. If it doesn't work, Argo will just finish deletion. But what we could to is check if command exists. 

I'm not sure about adding this logic to pre-processor, there are 3 places where we create/add Helm Values for installation:
- [Workflow](https://github.com/neuro-inc/platform-apps/blob/e7a164cde03e96cc741f27e00ab75017f7411a30/resources/workflow.yaml#L272) in Platform Apps adds PostProcessor (outputs) envs to helm values 
- App-Types [RunPreprocessor](https://github.com/neuro-inc/app-types/blob/cb90075bf703231e8f2182fdeb983c9b4c1ed457/src/apolo_app_types/cli.py#L165) function adds `APOLO_PASSED_CONFIG` to helm values
- App-Types PreProcessor add app-specific helm values

If feels like the first one is where we should be adding values used by hooks